includes/Media.php

```diff
@@ -204,6 +207,112 @@ public function init() {
 		add_filter( 'wp_prepare_attachment_for_js', [ $this, 'wp_prepare_attachment_for_js' ], 10, 2 );
 
 		add_action( 'delete_attachment', [ $this, 'delete_video_poster' ] );
+
+		// Hide video posters from Media grid view.
+		add_filter( 'ajax_query_attachments_args', [ $this, 'filter_ajax_query_attachments_args' ] );
+		// Hide video posters from Media list view.
+		add_filter( 'pre_get_posts', [ $this, 'filter_poster_attachments' ] );
+		// Hide video posters from web-stories/v1/media REST API requests.
+		add_filter( 'rest_attachment_query', [ $this, 'filter_rest_poster_attachments' ], 10, 2 );
+	}
+
+	/**
+	 * Returns the tax query needed to exclude generated video poster images.
+	 *
+	 * @param array $args Existing WP_Query args.
+	 *
+	 * @return array  Tax query arg.
+	 */
+	private function get_poster_tax_query( array $args ) {
+		$tax_query = [
+			[
+				'taxonomy' => self::STORY_MEDIA_TAXONOMY,
+				'field'    => 'slug',
+				'terms'    => [ 'poster-generation' ],
+				'operator' => 'NOT IN',
+			],
+		];
+
+		/**
+		 *  Merge with existing tax query if needed,
+		 * in a nested way so WordPress will run them
+		 * with an 'AND' relation. Example:
+		 *
+		 * [
+		 *   'relation' => 'AND', // implicit.
+		 *   [ this query ],
+		 *   [ [ any ], [ existing ], [ tax queries] ]
+		 * ]
+		 */
+		if ( ! empty( $args['tax_query'] ) ) {
+			$tax_query[] = $args['tax_query'];
+		}
+
+		return $tax_query;
+	}
```

Comment:
This method looks / acts very similar to this 
https://github.com/google/web-stories-wp/blob/49adc48243db2d149c8623119ae63e838bbe971d/includes/REST_API/Stories_Media_Controller.php#L190-L210

Any chance it could be one method?